### DISCUSSION TITLE
A Web Interface

### DISCUSSION THREAD

**Discussion by Pthepeg** [GerryDazoo/Slinger#310] [Show and tell]
I built a web interface for the slingbox_server and for the remote control operation. I have slingbox_server running on a Raspberry Pi. Also running on the same Pi is a Wordpress installation to host the web interface. 

It's split into two pages: Slingserver, and Slingbox. Check out the screen captures from my phone.

![image](https://user-images.githubusercontent.com/115769485/202334327-94830919-0989-4a27-bdf0-f5881e6d506d.png)
The Slingserver page above lets me launch the slingbox_server at the chosen resolution. (Choosing a resolution effectively just chooses a different config file and restarts the slingbox_server) I play the stream in a remote location and the connection speed is limited. This makes it easy to  try for the best resolution but back it down until it works. The Status button runs a ps command to show process status and confirm slingbox_server process is running. The screen capture shows the output from the Log button.

![image](https://user-images.githubusercontent.com/115769485/202334445-e732d0d7-d530-48ee-bcb1-06f767c054db.png)
The Slingbox page above has the remote control buttons for the set top box. When launching slingbox_server, the config file specifies Remote=redirect.html, a simple file that redirects slingbox_server back to my Wordpress Slingbox page after receiving each remote control command. So I can modify my Slingbox page: add audio only, change the button arrangement, etc. and don't need to identify an updated html in the slingbox_server config file. To keep the remote control compact and fitting on my phone screen I eliminated all the numbered buttons and just use a drop down list to choose one of the available networks, then hit the Go button. (In the screen capture I've chosen TSN3 and it populated the channel field with 1022.)

Both pages have custom templates behind them where php launches slingbox_server, etc.

At the media player, I have only one URL:port bookmarked. All resolutions are available at the same bookmark. When I change resolutions using the web interface, I just relaunch the media player at the same URL:port.

I'm using a free ddns service, the web interface URL and network stream URL are accessible from anywhere. 

I think I like it better than the original Slingplayer controls :)
Thanks for some great work, Gerry.

**Comment by pfuller88** [comment]
great idea.  can you share the php code?

**Reply by Pthepeg** [reply]
Here's a few key php pieces: launching and killing slingbox_server.py, outputting the log and the process status


$commandStr = "/usr/bin/python3 -u /home/slingbox/slingbox_server.py";
$configStr = "/home/slingbox/configunified320.ini";
$logFile = "/tmp/slinglog.txt";
$searchStr = '['.$commandStr[0].']'.substr($commandStr, 1); //    [/]usr/bin/python3 -u /home/slingbox/slingbox_server.py

// launches slingbox_server.py in the background and directs the output to the logfile
shell_exec($commandStr.' '.$configStr.' 1> '.$logFile.'  2>&1 &');

// kills all instances of slingbox_server.py processes that were launched by the php code (user www-data)
shell_exec("kill $(ps aux | grep '$searchStr' | awk '{print $2}')");

// output the log file
echo shell_exec("cat $logFile");

// output process status for slingbox_server.py processes only
echo shell_exec("ps aux | grep '$searchStr'");

**Comment by pfuller88** [comment]
Awesome, thanks.
